media/js/newsletter/management.es6.js

```diff
@@ -0,0 +1,785 @@
+    'test-pilot',
+    'take-action-for-the-internet',
+    'knowledge-is-power'
+];
+
+const FXA_NEWSLETTERS_LOCALES = ['en', 'de', 'fr'];
+
+const UNSUB_UNSUBSCRIBED_ALL = 1;
+
+let _form;
+let _userData;
+let _newsletterData;
+let _stringData;
+
+/**
+ * Returns true if a given value is found in a given list of <select> options.
+ * @param {Array} options
+ * @param {String} value
+ * @returns {Boolean}
+ */
+function _hasOption(options, value) {
+    const index = options.findIndex((option) => option.value === value);
+
+    if (index !== -1) {
+        return true;
+    }
+
+    return false;
+}
+
+/**
+ * Sets the value of a <select> input to the given value if found.
+ * @param {Array} options
+ * @param {String} value
+ */
+function _setOption(options, value) {
+    const index = options.findIndex((option) => option.value === value);
+
+    if (index !== -1) {
+        options[index].selected = 'selected';
+    }
+}
+
+/**
+ * Convenience helper for window.fetch() that returns a Promise.
+ * @param {String} url
+ * @param {String} method
+ * @returns {Promise}
+ */
+function _fetch(url, method) {
+    return new window.Promise((resolve, reject) => {
+        window
+            .fetch(url, {
+                method: method,
+                headers: {
+                    Accept: 'application/json',
+                    'Content-Type': 'application/json'
+                }
+            })
+            .then((resp) => {
+                if (
+                    resp.statusText === 'OK' &&
+                    resp.status >= 200 &&
+                    resp.status < 300
+                ) {
+                    resolve(resp.json());
+                } else {
+                    reject(resp);
+                }
+            })
+            .catch((e) => {
+                reject(e);
+            });
+    });
+}
+
+/**
+ * Main newsletter management form object.
+ */
+const NewsletterManagementForm = {
+    meetsRequirements: () => {
+        return 'Promise' in window && 'fetch' in window;
+    },
+
+    getPageLocale: (locale) => {
+        return typeof locale !== 'undefined'
+            ? locale
+            : document.getElementsByTagName('html')[0].getAttribute('lang');
+    },
+
+    getPageURL: () => {
+        return window.location.href;
+    },
+
+    getUserEmail: () => {
+        return _userData ? _userData.email : null;
+    },
+
+    getFormCountry: () => {
+        return _form.querySelector('select[name="country"]').value;
+    },
+
+    getFormLang: () => {
+        return _form.querySelector('#id_lang').value;
+    },
+
+    /**
+     * Get an array of checked newsletter IDs from the management form.
+     * @returns {Array}
+     */
+    getCheckedNewsletters: () => {
+        return Array.from(
+            document.querySelectorAll(
+                '.newsletter-table .newsletter-checkbox:checked'
+            )
+        ).map((newsletter) => {
+            return `${newsletter.value}`;
```

Comment:
Is this a shortcut to coerce to a string? Otherwise you could do `.map((newsletter) => newsletter.value);` right?